src/server/protocol.ts

```diff
@@ -2633,6 +2633,21 @@ namespace ts.server.protocol {
         insertSpaceBeforeTypeAnnotation?: boolean;
     }
 
+    export interface Options {
+        readonly quote?: "double" | "single";
+        /**
+         * If enabled, TypeScript will search through all external modules' exports and add them to the completions list.
+         * This affects lone identifier completions but not completions on the right hand side of `obj.`.
+         */
+        readonly includeCompletionsForModuleExports?: boolean;
+        /**
+         * If enabled, the completion list will include completions with invalid identifier names.
+         * For those entries, The `insertText` and `replacementSpan` properties will be set to change from `.x` property access to `["x"]`.
+         */
+        readonly includeCompletionsWithInsertText?: boolean;
+        readonly importModuleSpecifierPreference?: "relative" | "baseUrl";

```

Comment:
We should reconsider if we want to call this something like "non-relative" just because most of the time, people are using this with path-mapping; but I think it really depends on the behavior. If we *always* give completions from the `baseUrl` (probably undesirable) then we can keep this as `baseUrl`. But if we already have this, it's probably fine. @mhegazy @sheetalkamat